src/Modules/UngradedModules.jl

```diff
@@ -0,0 +1,2689 @@
+mutable struct FreeMod{T} <: ModuleFP{T}
+  R::CRing
+  n::Int
+  S::Array{Symbol, 1}
+
+  ingoing_morphisms::Array{<:ModuleMap,1}
+  outgoing_morphisms::Array{<:ModuleMap,1}
+
+  AbstractAlgebra.@declare_other
+
+  function FreeMod(n,b::CRing,c)
+    r = new{elem_type(b)}()
+    r.n = n
+    r.R = b
+    r.S = c
+
+    r.ingoing_morphisms = Array{ModuleMap,1}()
+    r.outgoing_morphisms = Array{ModuleMap,1}()
+
+    return r
+  end
+end
+
+# if one des not provide names for the generators, the standard names e_i are used for the unit vectors
+function FreeMod(R::CRing, n::Int, name::String = "e"; cached::Bool = false) 
+  return FreeMod(n, R, [Symbol("$name[$i]") for i=1:n])
+end
+free_module(R::CRing, n::Int, name::String = "e"; cached::Bool = false) = FreeMod(R, n, name, cached = cached)
+
+#=XXX this cannot be as it is inherently ambigous
+  - FreeModule(R, n)
+  - direct sum of rings, ie. a ring
+  - set of n-th powers of R
+thus the "category" needs to be set explicitly
+
+^(R::Ring_dec, n::Int) = FreeModule(R, n)
+=#
+
+function AbstractAlgebra.extra_name(F::FreeMod)
+  return nothing
+end
+
+function (F::FreeMod)()
+  return FreeModuleElem(sparse_row(base_ring(F)), F)
+end
+
+function show(io::IO, F::FreeMod)
+  @show_name(io, F)
+  @show_special(io, F)
+
+  print(io, "Free module of rank $(F.n) over ")
+  print(IOContext(io, :compact =>true), F.R)
+#=
+  i = 1
+  while i < dim(F)
+    d = F.d[i]
+    j = 1
+    while i+j <= dim(F) && d == F.d[i+j]
+      j += 1
+    end
+    print(IOContext(io, :compact => true), F.R, "^$j")
+    print(IOContext(io, :compact => true), "(", -d, ")")
+    if i+j < dim(F)
+      print(io, " + ")
+    end
+    i += j
+  end
+  =#
+end
+
+# Dimension, rank and number of generators
+dim(F::FreeMod) = F.n
+rank(F::FreeMod) = F.n
+ngens(F::FreeMod) = dim(F)
+
+# two free modules are equal if the rank and the ring are
+function ==(F::FreeMod, G::FreeMod)
+  # TODO it this enough or e.g. stored morphisms also be considered?
+  return F.R == G.R && rank(F) == rank(G)
+end
+
+# the zero module is the module of rank zero
+function iszero(F::FreeMod)
+  return rank(F) == 0
+end
+
+# elements of free modules are encoded in SRows
+struct FreeModuleElem{T}
+  coords::SRow{T} # also usable via coeffs
+  parent::FreeMod{T}
+end
+
+function Base.getproperty(v::FreeModuleElem, s::Symbol)
+  if s == :coeffs
+    return getfield(v, :coords)
+  end
+  return getfield(v, s)  
+end
+
+function getindex(v::FreeModuleElem, i::Int)
+  if isempty(v.coords)
+    return zero(base_ring(v.parent))
+  end
+  return v.coords[i]
+end
+
+elem_type(::Type{FreeMod{T}}) where {T} = FreeModuleElem{T}
+parent_type(::Type{FreeModuleElem{T}}) where {T} = FreeMod{T}
+elem_type(::FreeMod{T}) where {T} = FreeModuleElem{T}
+
+function show(io::IO, e::FreeModuleElem)
+  if length(e.coords) == 0
+    print(io, 0)
+    return
+  end
+  i = 1
+  while i <= length(e.coords)
+    print(io, "(", e.coords.values[i], ")*", e.parent.S[e.coords.pos[i]])
```

Comment:
@fieker suggests to use string interpolation, i.e.:
```suggestion
    print(io, "($(e.coords.values[i]))*$(e.parent.S[e.coords.pos[i]])")
```
Whether this is better or not IMHO depends on the situation; here I find it both equally bad ;-).